activerecord/lib/active_record/autosave_association.rb

```diff
@@ -267,7 +267,7 @@ def destroyed_by_association
     # Returns whether or not this record has been changed in any way (including whether
     # any of its nested autosave associations are likewise changed)
     def changed_for_autosave?
-      new_record? || changed? || marked_for_destruction? || nested_records_changed_for_autosave?
+      changed? || marked_for_destruction? || nested_records_changed_for_autosave?
```

Comment:
I think I would not change this part. I am saying that because `marked_for_destructio?` is in here too. So I think this method should return what the comment describes: `whether or not this record has been changed in any way`